pkgs/native_assets_cli/lib/src/api/build_config.dart

```diff
@@ -32,168 +45,155 @@ abstract class BuildConfig {
   /// dependency of another. For this it is convenient to know the packageRoot.
   Uri get packageRoot;
 
-  /// The target being compiled for.
-  ///
-  /// Not available in [dryRun].
-  Target get target;
-
   /// The architecture being compiled for.
   ///
-  /// Not available in [dryRun].
-  Architecture get targetArchitecture;
+  /// Not specified (`null`) during a [dryRun].
+  Architecture? get targetArchitecture;
 
   /// The operating system being compiled for.
-  OS get targetOs;
+  OS get targetOS;
 
   /// When compiling for iOS, whether to target device or simulator.
   ///
-  /// Required when [targetOs] equals [OS.iOS].
+  /// Required when [targetOS] equals [OS.iOS].
   ///
-  /// Not available in [dryRun].s
+  /// Not available during a [dryRun].
   IOSSdk? get targetIOSSdk;
 
   /// When compiling for Android, the minimum Android SDK API version to that
   /// the compiled code will be compatible with.
   ///
-  /// Required when [targetOs] equals [OS.android].
+  /// Required when [targetOS] equals [OS.android].
   ///
-  /// Not available in [dryRun].
+  /// Not available during a [dryRun].
   ///
   /// For more information about the Android API version, refer to
   /// [`minSdkVersion`](https://developer.android.com/ndk/guides/sdk-versions#minsdkversion)
   /// in the Android documentation.
   int? get targetAndroidNdkApi;
 
-  /// Preferred linkMode method for library.
+  /// The preferred linkMode method for [CCodeAsset]s.
   LinkModePreference get linkModePreference;
 
-  /// Metadata from direct dependencies.
+  /// Metadata from a direct dependency.
   ///
-  /// The key in the map is the package name of the dependency.
+  /// The [packageName] of is the package name of the direct dependency.
   ///
-  /// The key in the nested map is the key for the metadata from the dependency.
+  /// Returns `null` if metadata was not provided.
   ///
-  /// Not available in [dryRun].
-  Map<String, Metadata>? get dependencyMetadata;
+  /// Not available during a [dryRun].
+  Object? metadatum(String packageName, String key);
 
   /// The configuration for invoking the C compiler.
   ///
-  /// Not available in [dryRun].
+  /// Not available during a [dryRun].
   CCompilerConfig get cCompiler;
 
-  /// Don't run the build, only report the native assets produced.
+  /// Whether the current run is a "dry run".
+  ///
+  /// If so, the build won't actually be run, but will report the native assets
+  /// which would have been produced.
   bool get dryRun;
```

Comment:
Should we consider making it `BuildSubCommand subCommand` with `BuildSubCommand.build` and `BuildSubCommand.list`?

E.g. coming up with a way to have subcommands for the cli of any hook?

Reply:
Adding subcommands seems somewhat heavy when we only have two. It requires branching with equality checks in places.
Also we couldn't share the subcommands between `build.dart` and `link.dart` later.

As for renaming, it would deviate from what Make and XCodeBuild are doing:

```
$ make --help
  -n, --just-print, --dry-run, --recon
                              Don't actually run any commands; just print them.
$ xcodebuild -h
    -dry-run                                                 do everything except actually running the commands
```

Subcommands would bring it closer to what Ninja is doing with "Extra tools" https://ninja-build.org/manual.html

Let's postpone doing subcommands until we have a need for it, and keep it dry-run for now.